products/views.py

```diff
@@ -1,3 +1,62 @@
-from django.shortcuts import render
+import datetime
 
-# Create your views here.
+from django.http  import JsonResponse
+from django.views import View
+
+from dr_martens.checkitem   import CheckItem
+from dr_martens.login_check import login_check
+from products.models        import (Product,
+                                    ProductImage,
+                                    ProductOption)
+
+class DetailView(View):
+    @login_check
+    def get(self, request, *args, **kwargs):
```

Comment:
```suggestion
    def get(self, request, product_id, *args, **kwargs):
```
product_id는 필수 인자이니까 이렇게 넣어주면 좋을 것 같네요.

Reply:
리뷰 감사합니다.

다만, product_id를 함수의 인자로 받게 되면 데코레이터의 인자도 변경되어야 할 것 같습니다.
아마 데코레이터는 ProductDetailView 클래스 뿐만 아니라 다른 클래스에서도 사용을 할 것 같은데, 이렇게 수정하면 사용이 불가능할 것 같은 생각이 듭니다.
따라서 현재 코드와 같이 키워드 인자로 받아도 괜찮지 않을까요?

Reply:
네 그래도 될 것 같네요.

그런데 detail api 꼭 로근인이 필요한 이유는 무엇일까요? 
당연히 토큰이 없을 때 ip를 얻는 과정이 있어 동작에는 문제가 없어 보이지만, 해당 api에 로그인 데코레이터를 다는 것에 의문이 듭ㄴ다

Reply:
사실 로그인 데코레이터 중에서 토큰 만료를 프론트엔드에게 알려주는 내용이 있어서 API에 데코레이터를 추가하였습니다.
불필요한 데코레이터라고 생각되어 제거하였습니다.
그러면 토큰 만료는 아래와 같은 로직으로 처리하여도 괜찮을까요?

1. 토큰 발급 시, 만료 시간도 같이 프론트엔드에게 전달
2. 만료 시간이 넘으면 프론트엔드에서 토큰 삭제(로그아웃)